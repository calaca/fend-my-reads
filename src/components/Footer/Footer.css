footer {
  background-color: #f1f1f1;
  padding: 20px 30px;
  text-align: center;
  box-shadow: 0px 0px 15px 0px rgba(0, 0, 0, 0.15);
  height: 60px;
  position: absolute;
  bottom: 0;
  width: 100%;
}

.footer-text,
.footer-link {
  color: #bdbdbd;
  font-size: 14px;
  font-weight: 200;
}

.footer-text {
  margin: 0;
}

.footer-link {
  text-decoration: none;
  transition: all 0.4s ease-in-out;
}

.footer-link:hover {
  color: #61dafb;
}

.footer-icon {
  height: 18px;
  width: auto;
  transform:rotate(0deg);
  animation: spin 3s linear infinite;
}

@keyframes spin {
  from {
    transform:rotate(0deg);
  }
  to {
    transform:rotate(360deg);
  }
}
