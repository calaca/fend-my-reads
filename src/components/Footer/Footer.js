import React from 'react';
import reactIcon from '../../icons/react-icon.svg';
import './Footer.css';

const Footer = () => {
  return (
    <footer>
      <p className="footer-text">made with <img className="footer-icon" src={reactIcon} alt="React" title="React" /> by <a className="footer-link" href="https://calaca.github.io">calaca</a></p>
    </footer>
  )
};

export default Footer;
