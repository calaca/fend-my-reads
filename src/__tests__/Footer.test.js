import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Footer from '../components/Footer/Footer';

describe('<Footer />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Footer />);
  });

  it('renders without crashing', () => {
    wrapper;
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a link to my portfolio page', () => {
    const url = 'https://calaca.github.io';
    const link = wrapper.find('.footer-link');
    expect(link.prop('href')).toEqual(url);
  });
});
