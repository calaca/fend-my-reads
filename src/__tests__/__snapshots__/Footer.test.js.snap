// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        className="footer-text"
>
        made with 
        <img
                alt="React"
                className="footer-icon"
                src="react-icon.svg"
                title="React"
        />
         by 
        <a
                className="footer-link"
                href="https://calaca.github.io"
        >
                calaca
        </a>
</p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "made with ",
          <img
            alt="React"
            className="footer-icon"
            src="react-icon.svg"
            title="React"
/>,
          " by ",
          <a
            className="footer-link"
            href="https://calaca.github.io"
>
            calaca
</a>,
        ],
        "className": "footer-text",
      },
      "ref": null,
      "rendered": Array [
        "made with ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "React",
            "className": "footer-icon",
            "src": "react-icon.svg",
            "title": "React",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        " by ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "calaca",
            "className": "footer-link",
            "href": "https://calaca.github.io",
          },
          "ref": null,
          "rendered": "calaca",
          "type": "a",
        },
      ],
      "type": "p",
    },
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="footer-text"
>
          made with 
          <img
                    alt="React"
                    className="footer-icon"
                    src="react-icon.svg"
                    title="React"
          />
           by 
          <a
                    className="footer-link"
                    href="https://calaca.github.io"
          >
                    calaca
          </a>
</p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "made with ",
            <img
              alt="React"
              className="footer-icon"
              src="react-icon.svg"
              title="React"
/>,
            " by ",
            <a
              className="footer-link"
              href="https://calaca.github.io"
>
              calaca
</a>,
          ],
          "className": "footer-text",
        },
        "ref": null,
        "rendered": Array [
          "made with ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "React",
              "className": "footer-icon",
              "src": "react-icon.svg",
              "title": "React",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          " by ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "calaca",
              "className": "footer-link",
              "href": "https://calaca.github.io",
            },
            "ref": null,
            "rendered": "calaca",
            "type": "a",
          },
        ],
        "type": "p",
      },
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
